### PR TITLE
[BEAM-2562] Add integration test for logical operators

### PR DESCRIPTION
**Pull request by XuMingmin**
R: @xumingming @takidau  @jbonofre

### CODE REVIEW

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlRowCoder.java [hunk 1]
```diff
@@ -49,6 +50,8 @@
   private static final DoubleCoder doubleCoder = DoubleCoder.of();
   private static final InstantCoder instantCoder = InstantCoder.of();
   private static final BigDecimalCoder bigDecimalCoder = BigDecimalCoder.of();
+  private static final SerializableCoder<Boolean> booleanCoder =
```
Comment (xumingming) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlRowCoder.java]: I suspect a `ByteCoder` is better than `SerializableCoder` in terms of performance and deterministcity?

**Comment by xumingming** [comment]
Can you also update the test: `BeamSqlRowCoderTest` to include `boolean`?

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlRowCoder.java (hunk 1 above)
Comment (XuMingmin) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlRowCoder.java]: make sense especially for deterministcity, would be a problem with SerializableCoder if it's part of GROUP_BY.

**Comment by XuMingmin** [comment]
@takidau @jbonofre @kennknowles could you help to merge this PR?

**Comment by JingsongLi** [comment]
Merged.

**Comment by xumingming** [comment]
Thanks @JingsongLi !

**Comment by XuMingmin** [comment]
Thank you @JingsongLi @xumingming

**Pull request closed without merging**